examples/trigger-phrase/agent.py

```diff
@@ -0,0 +1,166 @@
+import asyncio
+import logging
+from typing import AsyncIterable, Optional
+
+from dotenv import load_dotenv
+from livekit import rtc
+from livekit.agents import (
+    AutoSubscribe,
+    JobContext,
+    WorkerOptions,
+    cli,
+    stt,
+    tokenize,
+    tts,
+    llm,
+)
+from livekit.plugins import deepgram, openai, silero, elevenlabs
+
+load_dotenv()
+
+logger = logging.getLogger("my-worker")
+logger.setLevel(logging.INFO)
+
+word_tokenizer_without_punctuation: tokenize.WordTokenizer = (
+    tokenize.basic.WordTokenizer(ignore_punctuation=True)
+)
+
+trigger_phrase = "Hi Bob!"
+trigger_phrase_words = word_tokenizer_without_punctuation.tokenize(text=trigger_phrase)
+
+
+async def _playout_task(
+    playout_q: asyncio.Queue, audio_source: rtc.AudioSource
+) -> None:
+    # Playout audio frames from the queue to the audio source
+    while True:
+        frame = await playout_q.get()
+        if frame is None:
+            break
+
+        await audio_source.capture_frame(frame)
+
+
+async def _respond_to_user(
+    stt_stream: stt.SpeechStream,
+    tts: tts.TTS,
+    agent_audio_source: rtc.AudioSource,
+    local_llm: llm.LLM,
+    llm_stream: llm.LLMStream,
+):
+    playout_q = asyncio.Queue[Optional[rtc.AudioFrame]]()
+    tts_stream = tts.stream()
+
+    async def _synth_task():
+        async for ev in tts_stream:
+            playout_q.put_nowait(ev.frame)
+
+        playout_q.put_nowait(None)
+
+    synth_task = asyncio.create_task(_synth_task())
+    playout_task = asyncio.create_task(_playout_task(playout_q, agent_audio_source))
+
+    async for ev in stt_stream:
+        if ev.type == stt.SpeechEventType.FINAL_TRANSCRIPT:
+
+            new_transcribed_text_words = word_tokenizer_without_punctuation.tokenize(
+                text=ev.alternatives[0].text
+            )
+            for i in range(len(trigger_phrase_words)):
+                if (
+                    len(new_transcribed_text_words) < len(trigger_phrase_words)
+                    or trigger_phrase_words[i].lower()
+                    != new_transcribed_text_words[i].lower()
+                ):
+                    # ignore user speech by not sending it to LLM
+                    break
+                elif i == len(trigger_phrase_words) - 1:
+                    # trigger phrase is validated
+                    new_chat_context = llm_stream.chat_ctx.append(
+                        text=ev.alternatives[0].text
+                    )
+                    llm_stream = local_llm.chat(chat_ctx=new_chat_context)
+                    llm_reply_stream = _llm_stream_to_str_iterable(llm_stream)
+                    async for seg in llm_reply_stream:
+                        tts_stream.push_text(seg)
+                    tts_stream.flush()
```

Comment:
what happens if neither of these if statements are true?

Reply:
it would go to the next word in the `new_transcribed_text_words` and continues until all the words in `trigger_phrase_words` are validated, at least that's my assumption. please let me know if I missed something in that logic!